### PR TITLE
Fix bug in memset optimization of ranges::uninitialized_value_construct

### PR DESCRIPTION
**Pull request by miscco**
🤦 

memset takes count in bytes not number of elements...

### CODE REVIEW

File: tests/std/tests/P0896R4_ranges_alg_uninitialized_fill/test.cpp [hunk 1]
```diff
@@ -141,6 +141,20 @@ struct throwing_test {
     }
 };
 
+struct memset_test {
+    static constexpr int expected[] = {42, 42, 42};
```
Comment (StephanTLavavej) [inline tests/std/tests/P0896R4_ranges_alg_uninitialized_fill/test.cpp:145]: This isn't exercising `memset`. We have tests above that do: `// Validate int is properly converted to bool` tests filling 1-byte elements (`bool`) with an arbitrary value (`5` converted to `true`), and `// Validate zero-ing` tests filling 4-byte `int` with `0`.

The `int` `0` test is basically comprehensive (and would have caught the bug here), but `bool` is weird, so should we replace this new test with a 1-byte elements, like filling `unsigned char` with `42`?

File: tests/std/tests/P0896R4_ranges_alg_uninitialized_fill_n/test.cpp [hunk 2]
```diff
@@ -120,6 +120,20 @@ struct throwing_test {
     }
 };
 
+struct memset_test {
+    static constexpr int expected[] = {42, 42, 42};
```
Comment (StephanTLavavej) [inline tests/std/tests/P0896R4_ranges_alg_uninitialized_fill_n/test.cpp:124]: Same comment, change `int` to `unsigned char` or another 1-byte type here to exercise `memset`.

**Review by StephanTLavavej — CHANGES_REQUESTED** [review]

**Review by CaseyCarter — COMMENTED** [review]
LGTM modulo STL's comments.

**Review by StephanTLavavej — APPROVED** [review]

**Review by miscco — COMMENTED** [review]
Thanks for fixing the remaining issues

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for squashing this bug! This will ship in VS 2019 16.10 Preview 2. :beetle: :boom: :heart_eyes_cat:

**Pull request merged**